firmware/library/L5_Testing/factory_test.hpp

```diff
@@ -0,0 +1,229 @@
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/i2c.hpp"
+#include "L1_Drivers/ssp.hpp"
+#include "L2_Utilities/log.hpp"
+#include "L2_Utilities/status.hpp"
+#include "L3_HAL/onboard_led.hpp"
+#include "L3_HAL/ssd1306.hpp"
+
+
+
+class FactoryTest
+{
+  public:
+    static constexpr uint8_t kAccelerometerAddress = 0x1C;
+    static constexpr uint8_t kGestureAddress       = 0x39;
+    static constexpr uint8_t kTemperatureAddress   = 0x31;
+
+    void RunFactoryTest()
+    {
+      OledTest();
+      bool i2c_pass   = I2cTest();
+      bool flash_test = ExternalFlashTest();
+      OnBoardLedTest(i2c_pass, flash_test);
+    }
+
+  private:
+    void OledTest()
+    {
+      Ssd1306 display;
+      LOG_INFO("Starting OLED Hardware Test...");
+
+      LOG_INFO("Initializing OLED Hardware Test...");
+      display.Initialize();
+
+      LOG_INFO("Clearing Screen...");
+      display.Clear();
+
+      LOG_INFO("Clearing Screen finished.");
+
+      display.Update();
+      LOG_INFO("Inverting screen...");
+      display.InvertScreenColor();
+    }
+
+    bool ExternalFlashTest()
+    {
+      printf("++++++++++++++++++++++++++++++++++++++\n\n");
+      printf("Starting External Flash Test...\n\n");
+      Ssp ssp2(Ssp::Peripheral::kSsp2);
+      Gpio cs(1, 10);
+      bool result = false;
+      cs.SetAsOutput();
+      cs.SetHigh();
+
+      ssp2.SetPeripheralMode(Ssp::MasterSlaveMode::kMaster, Ssp::FrameMode::kSpi,
+                             Ssp::DataSize::kEight);
+      ssp2.SetClock(false, false, 1, 48);
```

Comment:
Lets set this clock so to something much slower like:
```suggestion
      ssp2.SetClock(false, false, 100, 48);
```